### ISSUE TITLE
Custom shapes and collections like ExcaliDraw

### ISSUE THREAD

**Issue by leochras** [xournalpp/xournalpp#3914]
**Is your feature request related to a problem? Please describe.**

No.

**Describe the solution you'd like**

In [ExcaliDraw](https://github.com/excalidraw),  I can save custom shapes for re-use. There are also collections for different topics (for example electronics). I think it would be great to integrate that into xournal.

**Describe alternatives you've considered**

The only alternative I can think of is copy and pasting the shape every time.

**Additional context**

Here's what it looks like in Excalidraw:
![image](https://user-images.githubusercontent.com/16163571/158314994-75a218f3-9642-4a26-a4bd-aab1db87aef9.png)

**Comment by rolandlo** [comment]
That looks useful indeed. Compare with this [issue post](https://github.com/xournalpp/xournalpp/issues/2122#issuecomment-662676480).

**Comment by K4LCIFER** [comment]
I almost feel like these sorts of things should be out of the scope of what this project should be. In my eyes, Xournal++ should only be regarded as an annotation/note taking application. I don't think that it's built for constructings accurate diagrams/schematics that these sorts of things are usually used for. The issue that I see with this is that it has a huge potential for scope creep. It might just end up adding a lot of unnecessary bloat.

There are already great programs out there that are specifically designed for this purpose like libreoffice draw and draw.io.

**Comment by leochras** [comment]
You're right, maybe this is more of a plugin idea. Is there a label for these sort of ideas?

**Comment by brettmelbourne** [comment]
I think this would be great! For taking notes, it's probably the case that everyone has shapes specific to their area of interest that are continually useful (like you want to use them in almost every note) but hard to draw well or clearly with a stylus (I'm bad at drawing!). It could save time in note taking to have a personal library. For me it's braces that I use extensively to annotate hand-written equations. To be able to create the shape once, put it in a library, and pull from the toolbar would be awesome!

**Comment by rolandlo** [comment]
See https://github.com/xournalpp/xournalpp/pull/6020
